sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala

```diff
@@ -97,6 +97,38 @@ class CodeGenerationSuite extends SparkFunSuite with ExpressionEvalHelper {
     assert(actual(0) == cases)
   }
 
+  test("SPARK-18091: split large if expressions into blocks due to JVM code size limit") {
+    val row = create_row("afafFAFFsqcategory2dadDADcategory8sasasadscategory24", 0)
```

Comment:
This sort of simulates the scenario I faced which surfaced this issue. I tried a couple of other simple ways but those couldn't reproduce the issue. Since if expression's generated code will be large when it's constituents (i.e. predicate, true value, false value) will be large I thought I'll go with a nested recursive testcase consisting of if expressions